client_youtube.py

```diff
@@ -0,0 +1,161 @@
+import os
+import pathlib
+import re
+import shutil
+from glob import glob
+from typing import List, Optional
+
+import youtube_dl
+from models import GenerateVideoTask
+
+
+class SubtitleRange:
+    def __init__(self, text: str, time_start: str, time_end: str):
+        self.text: str = text
+        self.time_start: str = time_start
+        self.time_end: str = time_end
+
+        # Injected later when the picture and audio are produced.
+        self.path_to_picture: Optional[str] = None
+        self.path_to_audio: Optional[str] = None
+
+    def add_paths_to_picture_and_audio(
+        self, path_to_picture: str, path_to_audio: str
+    ):
+        assert os.path.isfile(path_to_picture), path_to_picture
+        assert os.path.isfile(path_to_audio), path_to_audio
+        self.path_to_picture = path_to_picture
+        self.path_to_audio = path_to_audio
+
+
+class YouTubeDownloadResult:
+    def __init__(
+        self,
+        video_title: str,
+        subtitles: List[SubtitleRange],
+        path_to_video: str,
+        path_to_subtitles_file: str,
+    ):
+        assert isinstance(video_title, str), video_title
+        assert isinstance(subtitles, list), subtitles
+        assert isinstance(path_to_video, str), path_to_video
+        assert isinstance(path_to_subtitles_file, str), path_to_subtitles_file
+        assert os.path.isfile(path_to_video), path_to_video
+        assert os.path.isfile(path_to_subtitles_file), path_to_subtitles_file
+        self.video_title: str = video_title
+        self.subtitles: List[SubtitleRange] = subtitles
+        self.path_to_video: str = path_to_video
+        self.path_to_subtitles_file: str = path_to_subtitles_file
+
+
+class YouTubeClient:
+    @staticmethod
+    def download_video_files(video_task: GenerateVideoTask) -> YouTubeDownloadResult:
+        print(
+            f"yt2srs: YouTubeClient: downloading video: "
+            f"{video_task.youtube_video_url}"
+        )
+
+        YouTubeClient._download_subtitles(video_task=video_task)
+        video_info = YouTubeClient._download_video(video_task=video_task)
+        title = video_info["title"]
+
+        print(f"yt2srs: YouTubeClient: " f"downloaded video: {title}")
```

Comment:
This print lines are super useful when developing but should be deactivated in the extension. Any thoughts how to maintain this?

Reply:
I think we could maybe use the [builtin logging library](https://docs.python.org/3/library/logging.html), so we could control output level either programatically or using environment variables.